src/types/filedRequest.ts

```diff
@@ -0,0 +1,34 @@
+import {Insight} from "./insight";
+import {User} from "./user";
+import {Group} from "./group";
+
+export const REQUESTS_COLLECTION = "requests"
+
+export interface FiledRequest {
+    createdBy: User;
+    createDate: number;
+    title: string;
+    assignee: User | Group;
+    priority: RequestPriority;
+    status: RequestStatus;
+    deadline: number;
+    assignedDate?: number;
+    completedDate?: number;
+    project?: string; // TODO Add project type
+    description?: string;
+    comments?: string[];
+    insights? : Insight[]
+}
+
+enum RequestPriority {
+    Low,
+    Medium,
+    High
+}
```

Comment:
Could we replace enunms with types? e.g. ``` type RequestPriority = 'Low' | 'Medium' | 'High'  ```

Reply:
Technically I think you can do this directly in interfaces now but I think it helps with organization